src/unstract/sdk/llm.py

```diff
@@ -69,15 +67,24 @@ def _initialise(self):
     def complete(
         self,
         prompt: str,
-        retries: int = 3,
+        process_text: Optional[Callable[[str], str]] = None,
         **kwargs: Any,
     ) -> Optional[dict[str, Any]]:
         try:
             response: CompletionResponse = self._llm_instance.complete(prompt, **kwargs)
+            process_text_output = {}
+            if process_text:
+                try:
+                    process_text_output = process_text(response, LLM.json_regex)
```

Comment:
What can be candidates for  process_text() @Deepak-Kesavan  ?

Reply:
I suggest adding it as a docstring

Reply:
Added docstring